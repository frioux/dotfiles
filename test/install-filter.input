#!/bin/bash

dir='test/,install-filter'
file_asis="test/,install-filter.asis"
file_filtered="test/,install-filter.filtered"

echo '@@VERSION@@' >"$file_asis"
chmod 700 "$file_asis"
echo '@@VERSION@@' >"$file_filtered"
chmod 700 "$file_filtered"

make --always-make >/dev/null \
  "INSTALLATION_DIR=$dir" \
  "TARGETS_STATIC=$file_asis $file_filtered" \
  'SHOULD_INSTALL_ASIS_P=$(filter %.asis,$(1))' \
  install

find "$dir"
diff "$file_asis" "$dir/$file_asis" >/dev/null; echo "asis $?"
test -x "$dir/$file_asis"; echo "asis executable $?"
diff "$file_filtered" "$dir/$file_filtered" >/dev/null; echo "filtered $?"
test -x "$dir/$file_filtered"; echo "filtered executable $?"

rm -rf "$dir" "$file_asis" "$file_filtered"

# __END__
