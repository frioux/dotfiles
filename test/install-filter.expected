test/,install-filter
test/,install-filter/test
test/,install-filter/test/,install-filter.asis
test/,install-filter/test/,install-filter.filtered
asis 0
asis executable 0
filtered 1
filtered executable 0
